keps/sig-network/2157-kube-proxy-config/README.md

```diff
@@ -0,0 +1,207 @@
+
+We thus propose a few simple consistency amendments to the input handling of the kube-proxy which will
+be self-consistent and prevent "no-op" inputs that are silently ignored, and in general, suggest transparency
+in terms of what inputs to kube-proxy are being utilized. The KEP intends to enhance the visibility of the flags already deprecated and provides consistency between flags and
+the existent component configuration API in kube-proxy as stated on [KEP 783](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/wgs/783-component-base).
+
+## Motivation
+
+The latter command ALSO ignores correct flags `--proxy-mode=iptables`, as you might expect,
+meaning that users might try to configure something and then spend several minutes trying to figure out why
+their config wasn't exercised. Since there are several kube-proxy options that might be overridden (there are ~50 options configured in a typical
+config-map for the kube proxy), this is obviously a compounding problem.
+
+The lack of flag precedence over the configuration API, option defaulting, and validation with proper round trip
+test contributes to increasing the existent or newly added flags' inconsistency. These statements are a requirement for the
+future configuration API graduation covered on [KEP 784](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/784-kube-proxy-component-config).
+
+## Goals
+
+We thus suggest the following modifications to kube-proxy input handling.
+
+1) For some options (such as the `mode`) the configuration is printed.  We suggest that all options be printed after
+   log level 5, this is done, for example, already for the `mode` option. The configuration struct must be printed
+   as well. This can be used to debug the flags usage by the users, currently options like proxy-mode are not processed
+   at all via command line, EVEN IF there is nothing in the configMap, an example of configuration output:
+
+```
+I1119 16:23:58.582572       1 server_others.go:186] Using iptables Proxier.
+```
+
+2) Wrong command line option must be explicitly rejected (even if only ignored before).
+
+3) Flags must take precedence over configuration options.
+
+## Non-Goals
+
+- Add new command line options for the kube-proxy or fix logical issues in the way kube-proxy manages input for domain specific inputs.
+- Override all config map options in flags.
+- Graduate the kube-proxy configuration API to v1beta1 or GA, besides paving this path.
+- Implement platform or instance (local) specific configuration over the already existent API.
+- Any kind of dynamic-reload, of any sort.  It was decided in a recent sig-network call that the Kube-proxy should never do this.
+
+## Proposal
+
+We propose to make the codebase self-consistent WRT the goals above, which largely is self-explanatory:
+
+- We will print out all inputs from the kube-proxy config map
```

Comment:
One thing that might mislead here is that actually, it's not the ConfigMap that have the priority (because kube-proxy does not read the ConfigMap), but the config file :)

The standard daemonset created by kubeadm creates a volume based on this ConfigMap, and uses this based on "--config=/var/lib/kube-proxy/config.conf"

So we might probably say that we want to:
* Make some configuration selection based on file or args selection
* Say that we don't want to propose the evolution of kube-proxy Config file in this KEP, but that this needs some attention :)

Reply:
yes, makes it less confusing to mention the configuration file instead, since it's not required to be configmap (even if this will be the case most of the time).

- There will be no field selection, instead ALL flags take precedence when they exist until they are finally deprecated. 
- This is covered on Non-goals items: https://github.com/kubernetes/enhancements/pull/2158/files#diff-0d42cc0b7f3d4b89cbc19a12402de0bf8b825be5a132e16766c80215455903d1R82